src/openforms/prefill/contrib/klantinteracties/plugin.py

```diff
@@ -0,0 +1,93 @@
+from django.urls import reverse
+from django.utils.translation import gettext_lazy as _
+
+import requests
+import structlog
+
+from openforms.authentication.service import AuthAttribute
+from openforms.contrib.klantinteracties.client import get_klantinteracties_client
+from openforms.contrib.klantinteracties.models import KlantinteractiesConfig
+from openforms.plugins.exceptions import InvalidPluginConfiguration
+from openforms.submissions.models import Submission, SubmissionValueVariable
+from openforms.typing import JSONEncodable
+
+from ...base import BasePlugin
+from ...constants import IdentifierRoles
+from ...exceptions import PrefillSkipped
+from ...registry import register
+from .config import KlantInteractiesOptionsSerializer
+from .typing import KlantInteractiesOptions
+from .utils import transform_digital_addresses
+
+logger = structlog.stdlib.get_logger(__name__)
+
+PLUGIN_IDENTIFIER = "klantinteracties"
+
+
+@register(PLUGIN_IDENTIFIER)
+class klantinteractiesPlugin(BasePlugin[KlantInteractiesOptions]):
+    verbose_name = _("Klantinteracties API")
+    requires_auth = (AuthAttribute.bsn,)
+    options = KlantInteractiesOptionsSerializer
+
+    @classmethod
+    def get_prefill_values_from_options(
+        cls,
+        submission: Submission,
+        options: KlantInteractiesOptions,
+        submission_value_variable: SubmissionValueVariable,
+    ) -> dict[str, JSONEncodable]:
+        try:
+            client = get_klantinteracties_client()
+        except AssertionError:
+            logger.warning("klantinteracties_service_not_configured")
+            return {}
+
+        if not (
+            bsn_value := cls.get_identifier_value(
+                submission, identifier_role=IdentifierRoles.main
+            )
+        ):
+            logger.info(
+                "lookup_identifier_absent",
+                submission_uuid=str(submission.uuid),
+                plugin=cls,
+            )
+            raise PrefillSkipped("No BSN available.")
+
+        assert submission_value_variable.form_variable
+        variable = str(submission_value_variable.form_variable.key)
+
+        digital_addresses = client.get_digital_addresses_for_bsn(bsn=bsn_value)
```

Comment:
use the context manager form when using the client so that it properly and explicitly cleans up resources:

```py
with get_klantinteracties_client() as client:
    ...
```
this is the pattern that's used pretty much everywhere in the OF codebase